query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/filters/one_relation.rs

```diff
@@ -162,6 +162,60 @@ mod one_relation {
         Ok(())
     }
 
+    // nested to-one-relation read filters.
+    #[connector_test]
+    async fn nested_to_one_filter(runner: Runner) -> TestResult<()> {
+        test_data(&runner).await?;
+
+        insta::assert_snapshot!(
+            run_query!(&runner, r#"query { findManyBlog { name, post(where: { title: "post1" }) { title } }}"#),
+            @r###"{"data":{"findManyBlog":[{"name":"blog 1","post":null},{"name":"blog 2","post":null},{"name":"blog 3","post":null}]}}"###
+        );
+
+        insta::assert_snapshot!(
+            run_query!(&runner, r#"query { findManyBlog { name, post(where: { title: "post 1", comment: { is: { text: "comment 1" } } }) { title, comment { text } } }}"#),
+            @r###"{"data":{"findManyBlog":[{"name":"blog 1","post":{"title":"post 1","comment":{"text":"comment 1"}}},{"name":"blog 2","post":null},{"name":"blog 3","post":null}]}}"###
+        );
+
+        Ok(())
+    }
+
+    fn to_one_req() -> String {
+        let schema = indoc! {
+            r#"
+            model Blog {
+                #id(id, String, @id, @default(cuid()))
+                name   String
+                post   Post   @relation(fields: [postId], references: [id])
+                postId String @unique
+              }
+              
+              model Post {
+                #id(id, String, @id, @default(cuid()))
+                title      String
+                popularity Int
+                blog       Blog?
+              }
+              
+            "#
+        };
+
+        schema.to_owned()
+    }
+
+    // nested to-one-relation read filters.
+    #[connector_test(schema(to_one_req))]
+    async fn nested_req_to_one_filter_should_fail(runner: Runner) -> TestResult<()> {
+        assert_error!(
+            runner,
+            r#"query { findManyBlog { post(where: { title: "title 1" }) { title } } }"#,
+            2009,
+            "Failed to validate the query: `Argument does not exist on enclosing type."
+        );
+
+        Ok(())
```

Comment:
Would you mind to explain why this should error? FWIW the post has a title field on it. 

Why shouldn't it select those Blogs, which only existing post contain the title, and then project the title?

Reply:
The model post has a title field on it but since it's a required one-to-one, the `where` argument that I'm using in this query is not present in the schema at all